src/base/OpenMCCellAverageProblem.C

```diff
@@ -2403,12 +2406,9 @@ OpenMCCellAverageProblem::syncSolutions(ExternalProblem::Direction direction)
 
         // we need to then re-establish the data structures that map from OpenMC cells to the [Mesh]
         // (because the cells changed)
+        resetTallies();
         setupProblem();
       }
-#else
```

Comment:
Can you explain where this went?

Reply:
This `#else` statement was the cause of the segmentation faults on CIVET, and more generally it resulted in memory leaks when Cardinal is built without DAGMC.

When the mesh is changing OpenMCCellAverageProblem needs to reset tallies on every timestep (after the first step), and so it calls `resetTallies()` at the beginning of the step. This is immediately followed by a call to `setupProblem()`. Afterwards, if the mesh is moving `setupProblem()` is called a second time in that `#else` statement resulting in an initialization of OpenMC tallies twice for each time they're reset. This causes a memory leak as we don't manage the extra OpenMC tally created on each step.

The reason why this caused a segmentation fault is a little more complicated and involves the copy of the mesh created in `MeshTally` containing only active elements. This mesh copy is deleted both when `resetTally()` and `initializeTally()` are called (it's a unique pointer). Since we've been creating two OpenMC tallies per timestep (by accident) the first of those tallies still thinks the mesh copy is valid, and so it tries to tally on datastructures that have had their memory freed. This behaviour is undefined and results in a segmentation fault.

This bug only occurs when Cardinal is compiled without DAGMC as `resetTallies()` is called again when skinning the geometry (the previous usecase for the `#ifdef`/`#else` statement), so nothing breaks.